codex-ui/src/vue/components/alert/Alert.types.ts

```diff
@@ -0,0 +1,83 @@
+import type { Ref } from 'vue';
+
+/**
+ * Various alert type
+ */
+export type Alertype = 'success' | 'error' | 'warning' | 'info' | 'default';
+
+/**
+ * position of alert, default position -> bottom-center
+ */
+export enum POSITION {
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  BOTTOM_CENTER = 'bottom-center'
+}
+
+/**
+ * alert configuration
+ */
+export interface AlertOptions {
+  /**
+   * Custom icon class to be used.
+   *
+   */
+  icon?: string;
+  /**
+   * content to be rendered
+   */
+  message?: string;
+  /**
+   *  Type of the alert.
+   *
+   *  Can be any of `(success, error, default, info, warning)`
+   */
+  type?: Alertype;
+  /**
+   *  Position of the alert on the screen.
+   *
+   * (bottom-center).
+   */
+  position?: POSITION.BOTTOM_CENTER;
+  /**
+   * How many milliseconds for the alert to be auto dismissed
+   */
+  timeout?: number;
+}
+
+export interface AlertInterface {
```

Comment:
lets move it to the `useAlert` file and rename to the `useAlertComposableState`

Reply:
effected